src/daft-core/src/kernels/search_sorted.rs

```diff
@@ -142,6 +142,69 @@ fn search_sorted_utf_array<O: Offset>(
     PrimitiveArray::<u64>::new(DataType::UInt64, results.into(), None)
 }
 
+fn search_sorted_boolean_array(
+    sorted_array: &BooleanArray,
+    keys: &BooleanArray,
+    input_reversed: bool,
+) -> PrimitiveArray<u64> {
+    let array_size = sorted_array.len();
+    let mut left = 0_usize;
+    let mut right = array_size;
+
```

Comment:
an optimization you can make: you should be able to precompute the index that 0 -> 1 and 1 -> Null. Then for the keys, you can just return the value of the index.